java-frontend/src/main/java/org/sonar/java/collections/SetUtils.java

```diff
@@ -47,4 +47,18 @@ private SetUtils() {
       .flatMap(Set::stream)
       .collect(Collectors.toSet());
   }
+
+  public static <T> Set<T> difference(Set<T> set1, Set<T> set2) {
+    return set1.stream()
```

Comment:
I'm not sure about this implementation. According to the documentation of Guava, why not go for:
```
Set<T> newSet1 = new HashSet<>(set1);
newSet1.removeAll(set2);
return newSet1;
```